compactor2/src/components/files_split/target_level_upgrade_split.rs

```diff
@@ -0,0 +1,525 @@
+use std::{
+    cmp::{max, min},
+    fmt::Display,
+};
+
+use data_types::{CompactionLevel, ParquetFile};
+
+use super::FilesSplit;
+
+#[derive(Debug)]
+/// Split files into `[files_to_compact]` and `[files_to_upgrade]`
+/// To have better and efficient compaction performance, eligible upgradable files
+/// should not be compacted but only need to update its compaction_level to the target_level
+pub struct TargetLevelUpgradeSplit {
+    size: u64,
```

Comment:
I think it would help to have doc comments here
```suggestion
    // Maximum desired file size (try and avoid compacting files above this size)
    size: u64,
```

Another thing that might help is to use the same consistent name as the config:

```suggestion
    max_desired_file_size_bytes: u64,
```